### PR TITLE
zeromq@4.1 4.1.6 (new formula)

### PR DESCRIPTION
**Pull request by fvioz**
Import formula zeromq 4.1 from homebrew/versions.  Latest version has some issues with ruby projects like https://github.com/chuckremes/ffi-rzmq-core/issues/15, 

This allow those users stay this this version until compatibility is added.

### CODE REVIEW

**Comment by fxcoudert** [comment]
We already have `zeromq` (which is currently 4.2), `zeromq@3.2` and `zeromq@4.0` (all of which see already very little use, like 1–2% of the latest version). The goal of having versioned formulas is not to support every version released, but to allow some support for a prior version when many users or our own software depends on it.

Right now, I don't think there is a demonstrated need for yet another versioned zeromq formula.

**Comment by fxcoudert** [comment]
Also, I don't see that homebrew-versions did ever have a `zeromq41` formula, so I don't see how this is an import from homebrew-versions.

**Comment by fvioz** [comment]
@fxcoudert My bad I fix the homebrew-versions, about this formula for ruby users https://rubygems.org/gems/ffi-rzmq-core only the version 4.1.x works, currently there is no support for 4.2 the latest.

**Comment by fxcoudert** [comment]
> only the version 4.1.x works, currently there is no support for 4.2 the latest

I guess this is an upstream decision they made, we cannot support every possible version… as I said, we already have 3.2 and 4.0

**Review by MikeMcQuaid — APPROVED** [review]
I'm OK with this given it's a listed version on their site: http://zeromq.org/intro:get-the-software

File: formula_renames.json [hunk 1]
```diff
@@ -168,5 +168,6 @@
   "vim74": "vim@7.4",
   "wires": "geckodriver",
   "zeromq32": "zeromq@3.2",
-  "zeromq40": "zeromq@4.0"
+  "zeromq40": "zeromq@4.0",
+  "zeromq41": "zeromq@4.1"
```
Comment (fxcoudert) [inline formula_renames.json]: This is incorrect, it is not a migration or formula rename as far as I can see…

File: Formula/zeromq@4.1.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    if build.with? "libsodium"
+      args << "--with-libsodium"
+    else
+      args << "--without-libsodium"
+    end
+
+    args << "--with-norm" if build.with? "norm"
+
+    system "./autogen.sh" if build.head?
```
Comment (fxcoudert) [inline Formula/zeromq@4.1.rb]: There is no head option here

File: Formula/zeromq@4.1.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    sha256 "8677f4e36376d84752b00a2c9801c1fe8519ee9db7b6df051c52d203f1e9575e" => :el_capitan
+    sha256 "d987614f7d62306b6a2c9f51133f5e393678d25f5c8705821b40aebe956347b2" => :yosemite
+  end
+
+  keg_only :versioned_formula
+
+  option "with-libpgm", "Build with PGM extension"
+  option "with-norm", "Build with NORM extension"
+
+  deprecated_option "with-pgm" => "with-libpgm"
```
Comment (JCount) [inline Formula/zeromq@4.1.rb]: A new formulae shouldn't have, or need, `deprecated_option` mappings as their purpose is to provide backwards compatibility with prior installations of that same formula.

File: Formula/zeromq@4.1.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  homepage "http://www.zeromq.org/"
+  url "https://github.com/zeromq/zeromq4-1/releases/download/v4.1.6/zeromq-4.1.6.tar.gz"
+  sha256 "02ebf60a43011e770799336365bcbce2eb85569e9b5f52aa0d8cc04672438a0a"
+
+  bottle do
+    cellar :any
+    sha256 "7d2de85d31cb632d0bb9aa94292c0023e2dd9b94849cf419124cd48540aaa4f8" => :sierra
+    sha256 "8677f4e36376d84752b00a2c9801c1fe8519ee9db7b6df051c52d203f1e9575e" => :el_capitan
+    sha256 "d987614f7d62306b6a2c9f51133f5e393678d25f5c8705821b40aebe956347b2" => :yosemite
+  end
```
Comment (fxcoudert) [inline Formula/zeromq@4.1.rb]: No `bottle` block for a new formula: these are added automatically when we merge.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Review by fxcoudert — APPROVED** [review]

**Comment by JCount** [comment]
Thank you very much for your first contribution to Homebrew, @fvioz!! 🎊

**Pull request closed without merging**